src/rest/src/main/java/org/geoserver/rest/util/IOUtils.java

```diff
@@ -764,9 +764,10 @@ public static void inflate(
 
         final Enumeration<? extends ZipEntry> entries = archive.entries();
         try {
+            File destDir = outputDirectory.dir();
             while (entries.hasMoreElements()) {
                 ZipEntry entry = entries.nextElement();
-
+                org.geoserver.util.IOUtils.getZipOutputFile(destDir, entry);
```

Comment:
This is just as a check that it doesn't have a harmful path?  A get method call where the return value isn't used is a bit odd otherwise.  If so maybe add a comment explaining it.

Reply:
I added a comment to that line.